specutils/io/asdf/schemas/spectrum1d-1.0.0.yaml

```diff
@@ -19,7 +18,7 @@ properties:
     description: |
       SpectralCoord that represents the spectral axis of the spectrum
     anyOf:
-      - $ref: "spectral_coord-1.0.0"
+      - tag: tag:astropy.org:specutils/spectra/spectral_axis-1.0.0
```

Comment:
@WilliamJamieson , can you please explain why there is a need to change `$ref` to `tag` here but not in `spectrum_list-1.0.0.yaml`? Thanks!

Reply:
There is no change from a $ref to a tag here.

The $ref that I removed was a local reference which pointed at the schema that was removed as part of this PR. Meaning this schema was broken by that removal. However, the removed schema was nearly identical to a schema in asdf-coordinates-schemas (I'll address the difference below). In fact, it duplicated the URI, which means one of the two versions would override the other. Which in general can cause confusion (no schemas should have the same URI). Indeed, looking at how ASDF loaded the schemas the asdf-coordinates-schema would always be discovered first meaning the procided here would never be used (except in the test generated by the pytest-asdf plug-in). The one difference was that it included a tag within the schema, which I will discuss why this is unnecessary below. 

Instead, what is going on here is I generated a new tag for an existing schema (multiple tags can point to the same schema), using the manifest. Moreover, the schema that tag points to is the one from asdf-coordinates schemas discussed above. 

In point of fact, this tag is "redundant" because the difference between a tag and a $ref, assuming the tag points to the same schema as the $ref, is that in addition addition to validating the data against the schema, the tag is on that sub tree is also checked. Note that the tag is also checked if the schema lists a tag; however, doing so is strongly discouraged because it prevents reuse of the schema by new tags (for new objects which have the same representation but different behaviors like the difference between SpectralAxis and SpectralCoord). That is it prevents the ability to have multiple tags pointing at the same schema, which is exactly what I did here to avoid unnecessary duplications.

To explicitly see the redundancy, observe the anyOf statement being applied to the list of statements This means any of the listed things is acceptable. Since the spectral_axis points to the schema which is referenced by the $ref on the next line. This means that spectral_axis tags would validate against the $ref anyways. As any tagged object (regardless of the tag itself) which satisfies the $ref will pass that line. Thus the tag added is purely of documentation value (this spectral axis tag is one possibility). Really, the spectralcood tag should be added for completeness. 

Note that in general, it is better to use a tag than a $ref in this situation because it makes explicit the object type(s) one expects, unless you are working with something which can accept a broad category of similar objects (in terms of how you actually write them down).

This means I would prefer to remove the $ref all together in this schema, but as this asdf extension was poorly maintained I fear existing files may fail to validate under these tighter restrictions. Meaning I did not do that (for now). Similarly, the spectral_list really ought to be of the tag for this schema not the $ref, but again the same caution is advised. 

I suggest that once this PR is merged, a second PR be opened. In this new PR new versions of the existing schemas be introduced (alongside a new manifest). These schemas would have the $ref to tag changes I discussed above. This would allow for reading/validating existing files via the old manifest, but strictly enforce things on any new files as ASDF will default to writing under the latest available tags/schema.

Reply:
Thank you very much for the very detailed explanation!